HIPS/hip-1.md

```diff
@@ -0,0 +1,54 @@
+---
+hip: <to be assigned>
```

Comment:
I'd rather drop the assignment and just have them be sequentially generated, with similar justification as above with the ERC20 -> EIP20 creating a nice convention.

Reply:
I'm not sure what you mean by sequentially generated? The file has to be named with the number by the author of the PR, so there could be 5 open PRs with the same number at the same time. Which one would win? They'd have to later update the number as someone gets a number committed before theirs (potentially update up to 4 times in this example). 

If we get a lot of proposals, we may have to move towards a folder structure later (like Kubernetes), and the sequence number would be harder to discern.

Numbers don't sort properly unless we agree upon leading zeros (look at EIPS).

Just numbers provides no context about when it was created and what that HIP is by just scanning the list of HIPs (again look at EIPS).

Reply:
Totally good points, appreciate it. I was prioritizing what it'd look like to browse, rather than use necessarily. Perhaps a date + short title makes the most sense. 

However, I would strongly advocate for some way that we can create a standard naming convention that maps to something the industry is familiar with, e.g. ERC20, TRC10, etc. 

Any suggestions on how to achieve?

Reply:
Rust rfc has them name it as <4 digit pr number>-short-description.md.

Or everyone could name as 0000-short-description.md and PR approver would pick next number right before merging.

Reply:
I've seen that last one used a few times and that makes sense to me. It keeps everything sequential while resolving your race condition concern.

Reply:
To clarify for my own sake - 

A user would submit a pull request with a file formatted `0000-<short-description>.md`

Once their HIP has been approved and is about to merge, the PR approver would grant them the next sequential number, and the user (submitting the HIP) would have to update the file they're checking in to match that number, e.g. `0001-example-template.md`

Is that correct?

Reply:
That's my understanding

Reply:
I think it would be good to give them #'s prior being approved to simplify the reference process

Reply:
Could you clarify how you would suggest doing so @publu?